tests/deployment_tests/gitlab.js

```diff
@@ -0,0 +1,201 @@
+const argv = require('minimist')(process.argv.slice(2));
+const request = require('request-promise-native');
+const simpleGit = require('simple-git/promise')();
+const { gitlab } = require('../../deploy/configs/gitlab/gitlab.json');
+
+const adminUserName = gitlab.username;
+const adminPassword = gitlab.password;
+const gitlabHostname = gitlab.hostname;
+
+async function getToken(user = adminUserName, pass = adminPassword) {
+  let body;
+  try {
+    body = await request.post(`https://${gitlabHostname}/api/v4/session/`, { json: { login: user, password: pass } });
+  } catch (error) {
+    throw new Error('Error in obtaining a private token');
+  }
+  return body.private_token;
+}
+
+async function getId(userName) {
+  let userId = -1;
+  try {
+    const token = await getToken();
+    const body = await request.get(`https://${gitlabHostname}/api/v4/users?private_token=${token}`);
+    const entries = Array.from(JSON.parse(body));
+    for (let i = 0; i < entries.length; i += 1) {
+      if (entries[i].username === userName) {
+        userId = entries[i].id;
+      }
+    }
+  } catch (error) {
+    // this error will happen when the https call to gitlab fails
+    throw new Error(`Error in obtaining ID of user ${userName}`);
+  }
+  return userId;
+}
+
+async function listUsers() {
+  let usernames = [];
+  try {
+    const token = await getToken();
+    const body = await request.get(`https://${gitlabHostname}/api/v4/users?private_token=${token}`);
+    usernames = (JSON.parse(body)).map(entry => entry.username);
```

Comment:
I know we use method chaining in promises; so JS community does not have any qualms about method chaining and hence no problems with violating the law of demeter. Please see this [stackoverflow discussion](https://stackoverflow.com/questions/1103985/method-chaining-why-is-it-a-good-practice-or-not/).

If we can avoid method chaining without introducing boilerplate code, then we should.